Close #388608.

In the extremely unlikely event that a log file to be rotated is going 
to be created group readable or writable, and the current group of the 
to-be-truncated logfile or the current user's group (for creation of the 
new logfile) is not the intended one, guard against the logfile being 
briefly readable or writable by the current user's group, or the group 
the logfile already had (and probably will have after the operation).

Thanks to Florian Zumbiehl <florz@gmx.de> for being zealously persistent.

Index: logrotate-3.7.9/logrotate.c
===================================================================
--- logrotate-3.7.9.orig/logrotate.c	2011-06-22 15:15:29.087530649 +0100
+++ logrotate-3.7.9/logrotate.c	2011-06-22 15:15:37.963764798 +0100
@@ -233,7 +233,7 @@
 {
     int fd;
 
-    fd = open(fileName, flags, sb->st_mode);
+    fd = open(fileName, flags, (S_IRUSR | S_IWUSR) & sb->st_mode);
     if (fd < 0) {
 	message(MESS_ERROR, "error creating output file %s: %s\n",
 		fileName, strerror(errno));
